src/test/integrationTests/userController.test.js

```diff
@@ -143,4 +143,137 @@ describe('Integration tests for the user controller', () => {
         .to.equal('Invalid request. All fields are required');
     });
   });
+  before('Get user token', async () => {
+    const response = await chai.request(app).post('/api/v1/auth/login')
+      .send({
+        email: 'demo1@demo.com',
+        password: 'password',
+      });
+    const { token, id } = response.body.data.userDetails;
+    describe('Test profile update endpoints', () => {
+      const newData = {
+        firstName: 'Kelechi',
+        lastName: 'Janet',
+        password: 'ejie565ewiw',
+        birthdate: '1/5/2018',
+        gender: 'female',
+        preferredLanguage: 'English',
+        preferredCurrency: 'USD',
+        profileImage: 'https://profileurl.com',
+        city: 'Enugu',
+        state: 'Enugu',
+        zip: '154587',
+        country: 'Nigeria',
+        role: 'SuperAdmin',
+        department: 'Accounting',
+        lineManager: 'Daniels',
+      };
```

Comment:
implementing my feedback about making the update logic more effective will not prevent listing all these columns.